django/contrib/admin/models.py

```diff
@@ -106,6 +125,20 @@ def __str__(self):
 
         return gettext("LogEntry Object")
 
+    def get_action_by_repr(self):
+        if self.action_by_repr:
+            return self.action_by_repr
+
+        if self.user:
+            action_by_repr = self.user.get_username()
+            if hasattr(self.user, 'get_full_name'):
+                full_name = self.user.get_full_name()
+                if full_name:
+                    action_by_repr = f'{action_by_repr} ({full_name})'
+            return action_by_repr
+
+        return gettext("Unknown, or deleted user")
```

Comment:
Alternatively, we could return an empty string. Or a question mark? But I think something a bit more descriptive could be helpful.